src/tools/wix/BinderFileManager.cs

```diff
@@ -297,7 +300,7 @@ public virtual bool CompareFiles(string targetFile, string updatedFile)
 
                         for (int i = 0; i < targetReadLength; ++i)
                         {
-                            if (targetBuffer[i] != updatedBuffer[i])
+                            if (0 != memcmp(targetBuffer, updatedBuffer, targetReadLength))
```

Comment:
Looping over bytes isn’t too bad.  For our payloads, this is significantly faster.  It’s inside the loop, but the blocks are 64k and we’re looping over blocks.

 

Certainly, looping over bytes is just fine.

 

--

John M. Cooper

Leavenworth, KS

 

 

 

From: Bob Arnson [mailto:notifications@github.com] 
Sent: Sunday, June 28, 2015 2:56 PM
To: wixtoolset/wix3
Cc: John Merryweather Cooper
Subject: Re: [wix3] WIXBUG 4791: (redone) Fix race in lit.exe/light.exe related to wixlibs and la… (#274)

 

In src/tools/wix/BinderFileManager.cs <https://github.com/wixtoolset/wix3/pull/274#discussion_r33426272> :

> @@ -297,7 +300,7 @@ public virtual bool CompareFiles(string targetFile, string updatedFile)
>  
>                          for (int i = 0; i < targetReadLength; ++i)
>                          {
> -                            if (targetBuffer[i] != updatedBuffer[i])
> +                            if (0 != memcmp(targetBuffer, updatedBuffer, targetReadLength))

This is still inside the loop...? I'd like to see profiling data that it's an improvement before taking in an msvcrt function via P/Invoke.

—
Reply to this email directly or view it on GitHub <https://github.com/wixtoolset/wix3/pull/274/files#r33426272> .  <https://github.com/notifications/beacon/AGu1PV6PN15g8NAdGwiRivzUzf-T8Jy-ks5oYEi8gaJpZM4FK_fk.gif> 



---
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus